pkg/services/hivelocity/device/device.go

```diff
@@ -328,19 +353,58 @@ func (s *Service) actionVerifyShutdown(ctx context.Context) actionResult {
 	// remove condition to reset the timer - we set the condition anyway again
 	conditions.Delete(s.scope.HivelocityCluster, infrav1.DeviceProvisioningSucceededCondition)
 
-	// TODO: make shut down API call
-
+	err = s.scope.HVClient.PowerOffDevice(ctx, deviceID)
+	if err != nil {
+		return actionError{err: fmt.Errorf("[actionDeviceProvisioned] PowerOffDevice failed: %w", err)}
+	}
 	conditions.MarkFalse(
 		s.scope.HivelocityCluster,
 		infrav1.DeviceProvisioningSucceededCondition,
 		infrav1.DeviceShutdownCalledReason,
 		clusterv1.ConditionSeverityInfo,
 		"device shut down has been triggered",
 	)
-
 	return actionContinue{delay: 30 * time.Second}
 }
 
+// Set permanent error with an appropriate label,
+// and go back to the state associate to associate with another device via actionGoBack.
+func (s *Service) setReloadingTooLongTag(ctx context.Context, deviceID int32, provisionCondition *clusterv1.Condition) actionResult {
+	device, err := s.scope.HVClient.GetDevice(ctx, deviceID)
+	if err != nil {
+		s.handleRateLimitExceeded(err, "GetDevice")
+		if errors.Is(err, hvclient.ErrDeviceNotFound) {
+
+			conditions.MarkFalse(
+				s.scope.HivelocityMachine,
+				infrav1.DeviceReadyCondition,
+				infrav1.DeviceNotFoundReason,
+				clusterv1.ConditionSeverityError,
+				fmt.Sprintf("device %d not found", device.DeviceId),
+			)
+			record.Warnf(s.scope.HivelocityMachine, "DeviceNotFound", "Hivelocity device not found")
+			s.scope.HivelocityMachine.SetFailure(capierrors.UpdateMachineError, infrav1.FailureMessageDeviceNotFound)
+			return actionComplete{}
+		}
+		return actionError{err: fmt.Errorf("failed to get associated device: %w", err)}
+	}
+	tags := append([]string(nil), device.Tags...)
+	tags = append(tags, fmt.Sprintf("permanent-error=reloading-since-%s",
+		provisionCondition.LastTransitionTime.Format(time.RFC3339)))
+	s.scope.HVClient.SetDeviceTags(ctx, deviceID, tags)
+
+	conditions.MarkFalse(
+		s.scope.HivelocityMachine,
+		infrav1.DeviceReadyCondition,
+		infrav1.DeviceReloadingTooLongReason,
+		clusterv1.ConditionSeverityError,
+		fmt.Sprintf("device %d reloading too long", device.DeviceId),
+	)
+	record.Warnf(s.scope.HivelocityMachine, "DeviceReloadingTooLong", "Hivelocity device reloading too long")
+	s.scope.HivelocityMachine.SetFailure(capierrors.UpdateMachineError, infrav1.FailureMessageDeviceReloadingTooLong)
```

Comment:
you don't want to set this failure, as this would mean the machine will be killed. However, we can just go back and associate it with the next machine, since nothing important happened yet.

Reply:
@janiskemper good point. I updated the code and the msg.